### PR TITLE
Custom video formats

### PR DESCRIPTION
**Pull request by deadbeef84**

### CODE REVIEW

File: src/core/video_format.cpp [hunk 1]
```diff
@@ -112,21 +208,32 @@ video_format_desc::video_format_desc(video_format     format,
...
 
-video_format_desc::video_format_desc(video_format format) { *this = format_descs.at(static_cast<int>(format)); }
+video_format_desc::video_format_desc()
+    : format(video_format::invalid)
+{
+    *this = video_format_repository::invalid();
+}
 
-video_format_desc::video_format_desc(const std::wstring& name)
+bool operator==(const video_format_desc& lhs, const video_format_desc& rhs)
```
Comment (ronag) [inline src/core/video_format.cpp:217]: You could make this to just look at all fields and skip all the "special" paths.

Comment (niklaspandersson) [reply src/core/video_format.cpp:217]: Good point. 👍  
This is just cherry-picked from upstream right now. You'll have to beef it out with @deadbeef84 whether we should deviate from upstream, fix it in upstream of just leave it.

**Comment by deadbeef84** [comment]
@ronag ready to merge?

**Comment by deadbeef84** [comment]
ping @ronag

**Review by ronag — APPROVED** [review]

**Pull request merged**